Foundational changes for Cypher chunking

petraselmer opened the pull request:
This must be merged simultaneously with https://github.com/neo-technology/neo4j-manual-modeling/pull/86

Commit:
Foundational changes for Cypher chunking

Commit:
Textual amendments to section landing pages

Commit:
Inline TOC for the clauses' landing page.

Commit:
Inline TOC for the functions' landing page; fixed errors and styling issues.

Commit:
Inline TOC for the remaining sections' landing pages

Commit:
Corrected link

jjaderberg commented on manual/cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/FunctionsTest.scala:
```diff
@@ -122,7 +122,7 @@ class FunctionsTest extends DocumentingTestBase {
   @Test def size() {
     testThis(
-      title = "size()",
+      title = "size() - list",
```
I believe the title controls the file name. With a title of "size() - list" the file name becomes "size---list.asciidoc", which means the include directive for this content fails. If you run a `find manual/cypher/cypher-docs/ -name "*size*"` you should see the generated files. Mutatis mutandis also the other "size" and the two "length" items.

jjaderberg commented on manual/cypher/cypher-docs/src/docs/dev/clause.asciidoc:
```diff
@@ -0,0 +1,157 @@
+[[query-clause]]
+= Clauses
+
+[discrete]
```
I don't think the AsciiDoc processor knows `[discrete]` so it chokes. Can we make do without it?

jjaderberg reviewed (changes requested):
Very nice!

Two trifles indicated need fixing:
- no `[discrete]` styles in AsciiDoc
- file names for include directives must agree with what is generated

There is a third after that which at a glance looks like a renegade null character, but I haven't quite figured it out yet.

Commit:
Replaced the `[discrete]` text

Commit:
Reverted function title changes

petraselmer commented:
@jjaderberg thanks for the comments. I have addressed the two issues you raised.

jjaderberg committed:
Conditional offsets and updated paths

The Asciidoc processor does not handle relative level offsets. While we
don't care about the outputs from the upstream docs build we do care to
keep it functioning, so let's set those level offsets for the neo4/neo4j
build, to make it happy.

Also update where some generated docs are put so includes work, and
tweak some of the offsets to account for structural changes.
